include/bitcoin/protocol/zmq/message.hpp

```diff
@@ -66,12 +66,31 @@ class BCP_API message
 
     /// Remove a part from the queue top, false if empty queue or invalid.
     bool dequeue();
-    bool dequeue(uint32_t& value);
     bool dequeue(data_chunk& value);
     bool dequeue(std::string& value);
     bool dequeue(hash_digest& value);
     bool dequeue(address& value);
 
+    template<typename T,
+             typename std::enable_if<std::is_integral<T>::value, int>::type ...>
```

Comment:
Actually, you should just remove the type constraint. We generally only use these when the type is otherwise under-constrained. As long as the type passes `from_little_endian_unsafe<T>` it will work, so it is fully-constrained. This is why `Unsigned` is used for `enqueue(...)` as it documents the constraint for the caller.

Reply:
Good point, a much better way to go, thanks!